scala-oauth2-core/src/main/scala/scalaoauth2/provider/GrantHandler.scala

```diff
@@ -20,7 +20,7 @@ trait GrantHandler {
    */
   def clientCredentialRequired = true
 
-  def handleRequest[U](request: AuthorizationRequest, authorizationHandler: AuthorizationHandler[U])(implicit ctx: ExecutionContext): Future[GrantHandlerResult[U]]
+  def handleRequest[U](validatedClientId: Option[String], request: AuthorizationRequest, authorizationHandler: AuthorizationHandler[U])(implicit ctx: ExecutionContext): Future[GrantHandlerResult[U]]
```

Comment:
Yes - I think this is better 👍 by changing it to `parseClientCredential` it makes it more clear that any time you call that method you are merely parsing the credential and have no idea of it's validity.  I think the variable name in the `GrandHandler#handleRequest` method should be something more like `maybeValidatedClientCred` to signify that it is more than just an option of parsed client cred - it has already been validated by the `AuthorizationHandler#validateClient` provided by the implementor of the library. 

I have updated the PR with these changes, let me know if you see something else you would like me to change. Thanks!